### ISSUE TITLE
Synthetics monitoring failing without logs

### ISSUE THREAD

**Issue by daniadamatti** [elastic/synthetics#774]
From time to time we're experiencing this weird behavior on our synthetic monitoring:

- Steps are not even loaded on uptime
- Log shows that step failed when executing "waitForLoadState" even though we don't explicitly use this method

Here's a printscreen:
![image](https://github.com/elastic/synthetics/assets/98332672/5b0a1b72-5e8e-4b50-bb87-a8f6ec6bc7ce)

Here is an example of a journey:
```
import { journey, step } from '@elastic/synthetics'
import '../../environment'
import { SignInPage } from '../../models'

journey({ name: 'sign-in', tags: ['accounts-auth'] }, options => {
  const { page } = options
  const signInPage = new SignInPage(page)

  step('sign in', async () => {
    await this.page.goto(this.getUrl())
    const login = email
    await this.page.fill('data-testid=login-email', login)
    await this.page.fill(
      'data-testid=login-password',
      password
    )
    await this.page.click('data-testid=login-submit')
    await this.page.waitForNavigation()
  })

  step('select product', async () => signInPage.selectMarketingProduct())
})
```

Here is an example of our heartbeat:
```
heartbeat.monitors:
    - type: browser
      name: rd-synthetics-pro-accounts-auth
      id: rd-synthetics-pro-accounts-auth
      schedule: "@every 3m"
      filter_journeys:
        tags:
          - "accounts-auth"
      source:
        local:
          path: "${SUITES_DIR}"

    fields:
      env: production

    tags:
      - production

```

Versions:
We use @elastic/synthetics: 1.0.0-beta.14 and run our synthetics tests in using docker image: docker.elastic.co/beats/heartbeat:7.15.0

**Comment by paulb-elastic** [comment]
Hi @daniadamatti there have been many improvements in Synthetics since 7.x and the beta version you are using.  In fact we [don't recommend](https://www.elastic.co/guide/en/observability/7.17/monitor-uptime-synthetics.html#monitoring-synthetics) using browser checks on 7.x.

With Synthetics going [generally available with 8.8](https://www.elastic.co/blog/whats-new-elastic-observability-8-8-0), are you able to try a newer version?  Looking at your configuration above, I'd suggest looking at [project monitors](https://www.elastic.co/guide/en/observability/current/synthetics-get-started-project.html).

**Comment by paulb-elastic** [comment]
Closing this as we're now in GA.
